src/st/sel_psp/cutscene.c

```diff
@@ -0,0 +1,1062 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+#include "../sel/sel.h"
+
+typedef struct {
+    /* 0x00 */ u8* scriptCur;         // ptr to dialogue next character
+    /* 0x04 */ s16 startX;            // starting x coord
+    /* 0x06 */ s16 nextLineY;         // next line y coord
+    /* 0x08 */ s16 startY;            // starting y coord
+    /* 0x0A */ s16 nextCharX;         // next char x coord
+    /* 0x0C */ s16 nextLineX;         // next line x coord
+    /* 0x0E */ u16 nextCharY;         // next char y coord
+    /* 0x10 */ u16 portraitAnimTimer; // portrait animation timer
+    /* 0x12 */ u16 unk12;             // unknown
+    /* 0x14 */ u16 clutIndex;         // CLUT index
+    /* 0x16 */ u16 nextCharTimer;     // timer to next character
+    /* 0x18 */ Primitive* prim;       // for dialogue graphics rendering
+    /* 0x1C */ s32 primIndex;         // primIndex: unk
+} Dialogue4;                          // size = 0x20
```

Comment:
I don't see an immediate way to avoid a new Dialogue struct, but add it to cutscene.h instead of here if it ends up being necessary.
I'm looking at this situation, starting with #2906 .